source/Calamari.AzureAppService.Tests/TargetDiscoveryBehaviourIntegrationTestFixture.cs

```diff
@@ -141,6 +153,8 @@ public async Task Execute_WebAppWithNonMatchingTags_CreatesNoTargets()
 
             await CreateOrUpdateTestWebApp(tags);
 
+            await Task.Delay(waitForAzureToProcessPeriod);
```

Comment:
Rather than a hard delay, can we use the Eventually semantics to keep checking for a period of time.

Reply:
Unfortunately we don't have the ShouldEventually utility in Calamari but yeah I can implement a crude version of it here.

Reply:
I have copied the ShouldEventually class and related utility classes to Calamari and massaged them to fit into the Calamari world so other tests can use them too - it might be worth using this for some of the flakey tests we have.